src/components/GAddressLookup/component.js

```diff
@@ -0,0 +1,271 @@
+
+    componentDidUpdate({ isScriptLoaded: prevIsScriptLoaded }) {
+        const { isScriptLoaded, isScriptLoadSucceed } = this.props;
+        if (!prevIsScriptLoaded && isScriptLoaded && isScriptLoadSucceed) {
+            this.initComponent();
+        }
+    }
+
+    getContainerClassName() {
+        const { className } = this.props;
+        return classnames('rainbow-google-address-lookup_container', className);
+    }
+
+    initComponent() {
+        this.autocompleteService = new window.google.maps.places.AutocompleteService();
+        this.setState({ initialized: true });
+    }
+
+    handleChange(option) {
+        const { onChange } = this.props;
+
+        if (!option) {
+            this.setState({
+                value: null,
+                selectedPlace: option,
+            });
+            onChange(option);
+            return;
+        }
+
+        const { places, suggestions } = this.state;
+        const value = places.length > 0 ? places[suggestions.indexOf(option)] : option;
+
+        this.setState({
+            value,
+            selectedPlace: {
+                label: option.label,
+                description: option.description,
+                icon: <SelectedLocationIcon />,
+            },
+        });
+
+        onChange(value);
+    }
+
+    handleSearch(value) {
+        const { initialized } = this.state;
+        if (!initialized) return;
+
+        if (value && this.state.searchValue !== value) {
+            this.setState({ searchValue: value });
+            this.searchGoogleMaps(value).then(
+                results => this.processSearchResults(results),
+                () => this.handleError(),
+            );
+        } else if (!value) {
+            this.clear();
+        }
+    }
+
+    searchGoogleMaps(needle) {
+        const { searchOptions } = this.props;
+        this.setState({ isSearching: true });
+
+        return new Promise((resolve, reject) => {
+            this.autocompleteService.getPlacePredictions(
+                {
+                    ...searchOptions,
+                    input: needle,
+                },
+                (predictions, status) => {
+                    this.setState({ isSearching: false });
+
+                    if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+                        reject(status);
+                    }
+
+                    resolve(predictions);
+                },
+            );
+        });
+    }
+
+    processSearchResults(results) {
+        this.setState({
+            places: results,
+            suggestions: [
+                {
+                    label: (
+                        <span>
+                            Search for: <b>&lsquo;{this.state.searchValue}&rsquo;</b>
+                        </span>
+                    ),
+                    icon: <SearchValueIcon />,
+                },
+            ].concat(results.map(place => formatSuggestionItem(place, true, <LocationItemIcon />))),
```

Comment:
use this as:
```
this.setState({
    places: results,
    suggestions: getSuggestions(results, searchValue),
});
```
and then put getSuggestions inside helpers folder and it is more easily to unit test:
```
function getSuggestions(results, searchValue) {
    return [
                getSearchForOption(searchValue),
            ].concat(results.map(place => formatSuggestionItem(place, true, <LocationItemIcon />)))
}
```
and use getSearchForOption since it is used twice:
```
function getSearchForOption(searchValue) {
    return {
                    label: (
                        <span>
                            Search for: <b>&lsquo;{this.state.searchValue}&rsquo;</b>
                        </span>
                    ),
                    icon: <SearchValueIcon />,
                };
}
```